### DISCUSSION TITLE
Average of around 50 seconds

### DISCUSSION THREAD

**Discussion by Spider1354** [mKeRix/room-assistant#1110] [Q&A] [answered]
Hi,
i am pretty happy with room assistant but i am wondering whether its possible to get around the average calculation (or lower it) when using ble dtection.
You can easy verify it by looking at a mqtt explorer. After a (re)start it sends one update and waits for around 50 seconds till the distance is sent. Afterwards if you are moving around you will notice that the distance is averaging by above mentioned timeframe. I tried various configurations but i am not able to lower the average calculation for the mqtt messages (distance and rssi).
I hooe somebody can help:)
Regards

**Comment by mKeRix** [comment, marked as answer]
I just added new config options that will allow you to control the Kalman filter used to do the noise reduction (`kalmanProcessNoise` and `kalmanMeasurementNoise`) - which is what I believe you meant with the average calculation. Those will be out later today. By default, room-assistant will send out distance updates as soon as it receives and processes a BLE advertisement packet from your device, unless you explicitly configure `updateFrequency` in the BLE integration.

**Reply by Spider1354** [reply]
Thats pretty nice and might be usefull for others too:)
I am happy to test this out. 
Can u tell the default value or the range which is allowed for those setting? I guess those will be tag override options?
Thanks for your work. Appreciate it

**Reply by mKeRix** [reply]
The new settings are now documented on the [website](https://www.room-assistant.io/integrations/bluetooth-low-energy.html#settings), along with their defaults. Both accept positive numbers. Easiest way to get started tuning is pretty much just trying to adjust the value and see if the behavior suits you. There also are more scientific approaches that could lead to even better results, those are described in a few articles and papers that can be found online.
